src/widgetastic_patternfly/__init__.py

```diff
@@ -2351,3 +2359,219 @@ def item_select(self, item, close=True):
+            :py:class:`list` complete data on chart
+        """
+        data = []
+        for el in self.browser.elements(self.RECT):
+            self.browser.move_to_element(el)
+            data.append(self.tooltip.read())
+        return data
+
+
+class SingleLineChart(SparkLineChart):
+    """Represents the Single Line Chart from Patternfly (Data Visualization).
+
+    Args:
+        id: id of SingleLineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        single_line_chart = SingleLineChart(id="singleLineChart")
+    """
+    X_AXIS = ".//*[contains(@class, 'c3-axis c3-axis-x')]/*[contains(@class, 'tick')]"
+    tooltip = Table(locator=".//div[contains(@class,'c3-tooltip-container')]/table")
+
+    @property
+    def _elements(self):
+        br = self.browser
+        return {
+            x.get_attribute("textContent"): el
+            for (x, el) in zip(br.elements(self.X_AXIS), br.elements(self.RECT))
+        }
+
+    def _get_data(self, elements):
+        data = {}
+        for el in elements:
+            self.tooltip.clear_cache()
+            self.browser.move_to_element(el)
+            raw_data = {row[0].text: row[1].text for row in self.tooltip.rows()}
+            if self.tooltip.headers:
+                tooltip_data = {self.tooltip.headers[0]: raw_data}
+            else:
+                # In the absence of a header, `:` separates x-axis points and values.
+                tooltip_data = {
+                    h[:-1] if (h[-1] == ":") else h: value for h, value in raw_data.items()
+                }
+            data.update(tooltip_data)
+        return data
+
+    def read(self):
+        """read all data on chart
+
+        Returns:
+            :py:class:`dict` complete data on chart
+        """
+        return self._get_data(self._elements.values())
+
+    def get_values(self, x_axis):
+        """data for specific x-axis point on chart
+
+        Args:
+            x_axis: x-axis point as per chart
+
+        Returns:
+            :py:class:`dict` data for selected timestamp
+        """
+        el = [self._elements.get(x_axis)]
+        return self._get_data(el)
+
+
+class LineChart(SingleLineChart):
+    """Represents the Line Chart having legends from Patternfly (Data Visualization).
+
+    Args:
+        id: id of LineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        line_chart = LineChart(id="lineChart")
+    """
+    LEGENDS = ".//*[contains(@class, 'c3-legend-item c3-legend-item-')]"
+
+    @property
+    def _legends(self):
+        return {self.browser.text(leg): leg for leg in self.browser.elements(self.LEGENDS)}
+
+    @property
+    def legends(self):
+        """ Get all available legends
+
+        Returns:
+            :py:class:`list` all available legends
+        """
+        return list(self._legends.keys())
+
+    def legend_is_displayed(self, leg):
```

Comment:
Would it be beneficial to have each legend be it's own class and then have each of these methods be defined within the `Legend` class? I.e. The `legends` property above would return a list of `Legend` objects that each have methods for (`is_displayed`, `hide`, `display`, and `read`). Then these aggregate methods could just act on each `Legend` object.

Reply:
This may add too much unnecessary abstraction/code however, so let me know if it doesn't make sense for this particular case.

Reply:
yup; LOC will increase :+1:  which not make any seance. In `WT` we are playing with `web-elements` so better to go with `methods` because legends count depends on `developer` approach. He can add multiple legends each time... we have to need initialize `class` and use methods which will be hectic. I would like to go with `method`.